src/include/main/query_result.h

```diff
@@ -6,9 +6,9 @@
 #include "common/arrow/arrow.h"
 #include "common/types/types.h"
 #include "kuzu_fwd.h"
+#include "main/database.h"
```

Comment:
Can you move `DatabaseLifeCycleManager` into its own header file and we can get rid of the include of `main/database.h`?

Reply:
Done